lib/shopping_cart.rb

```diff
@@ -0,0 +1,43 @@
+class ShoppingCart
+  attr_reader :products, :name, :capacity, :details, :total_in_cart
+
+  def initialize(name, capacity)
+    @name = name
+    @capacity = capacity
+    @products = []
+    @total_in_cart = 0
+  end
+
+  def add_product(product_added)
+    @products << product_added
+  end
+
+  def details
+    @details = Hash.new
+    @details[:name] = @name
+    @details[:capacity] = @capacity[0..1].to_i
+    @details
+  end
+
+  def total_number_of_products
+    @products.each do |product|
+      @total_in_cart += product.quantity
+    end
+    @total_in_cart
```

Comment:
This does not work if you get the total, add another product and check it again. Total number of products should not be a value that is stored.